activesupport/lib/active_support/callbacks.rb

```diff
@@ -693,14 +693,18 @@ def skip_callback(name, *filter_list, &block)
 
         __update_callbacks(name) do |target, chain|
           filters.each do |filter|
-            filter = chain.find {|c| c.matches?(type, filter) }
+            callback = chain.find {|c| c.matches?(type, filter) }
 
-            if filter && options.any?
-              new_filter = filter.merge_conditional_options(chain, if_option: options[:if], unless_option: options[:unless])
-              chain.insert(chain.index(filter), new_filter)
+            if !callback
+              warn "#{type.to_s.capitalize} #{name} callback #{filter.inspect} has not been defined"
             end
 
-            chain.delete(filter)
```

Comment:
Sorry, I don't understand. In both cases the delete is at the end.

The git diff isn't very clear, because I've renamed filter to callback (to avoid shadowing the loop variable for filter)